### PR TITLE
Fix download scripts

### PR DESCRIPTION
**Pull request by brian-avery**
Currently, the downloadIstioCtl and downloadIstioCandidate scripts are broken on release-1.6 due to the fact that the architecture was added to the filename. This PR adds the ability to specify the architecture and download the appropriate version.

### CODE REVIEW

**Comment by brian-avery** [comment]
```
ISTIO_VERSION=1.6.0-alpha.8 ./downloadIstioCandidate.sh
Downloading istio-1.6.0-alpha.8 from https://github.com/istio/istio/releases/download/1.6.0-alpha.8/istio-1.6.0-alpha.8-linux.tar.gz ...Failed.

 Trying with ISTIO_ARCH. Downloading istio-1.6.0-alpha.8 from https://github.com/istio/istio/releases/download/1.6.0-alpha.8/istio-1.6.0-alpha.8-linux-amd64.tar.gz ...


Unable to download Istio 1.6.0-alpha.8 at this moment!
Please verify the version you are trying to download.

```
```
./downloadIstioCandidate.sh 
Downloading istio-1.5.1 from https://github.com/istio/istio/releases/download/1.5.1/istio-1.5.1-linux.tar.gz ...
Istio 1.5.1 Download Complete!

Istio has been successfully downloaded into the istio-1.5.1 folder on your system.

Next Steps:
See https://istio.io/docs/setup/kubernetes/install/ to add Istio to your Kubernetes cluster.

To configure the istioctl client tool for your workstation,
add the /home/bavery/code/istio/src/istio/release/istio-1.5.1/bin directory to your environment path variable with:
         export PATH="$PATH:/home/bavery/code/istio/src/istio/release/istio-1.5.1/bin"

Begin the Istio pre-installation verification check by running:
         istioctl verify-install 

Need more information? Visit https://istio.io/docs/setup/kubernetes/install/ 
```

```
ISTIO_VERSION=1.6.0-alpha.2 ./downloadIstioCandidate.sh
Downloading istio-1.6.0-alpha.2 from https://github.com/istio/istio/releases/download/1.6.0-alpha.2/istio-1.6.0-alpha.2-linux.tar.gz ...Failed.

 Trying with ISTIO_ARCH. Downloading istio-1.6.0-alpha.2 from https://github.com/istio/istio/releases/download/1.6.0-alpha.2/istio-1.6.0-alpha.2-linux-amd64.tar.gz ...

Istio 1.6.0-alpha.2 Download Complete!

Istio has been successfully downloaded into the istio-1.6.0-alpha.2 folder on your system.

Next Steps:
See https://istio.io/docs/setup/kubernetes/install/ to add Istio to your Kubernetes cluster.

To configure the istioctl client tool for your workstation,
add the /home/bavery/code/istio/src/istio/release/istio-1.6.0-alpha.2/bin directory to your environment path variable with:
         export PATH="$PATH:/home/bavery/code/istio/src/istio/release/istio-1.6.0-alpha.2/bin"

Begin the Istio pre-installation verification check by running:
         istioctl verify-install 

Need more information? Visit https://istio.io/docs/setup/kubernetes/install/ 
```

**Comment by brian-avery** [comment]
```
ISTIO_VERSION=1.6.0-alpha.8 ./downloadIstioCtl.sh
Downloading istioctl-1.6.0-alpha.8 from https://github.com/istio/istio/releases/download/1.6.0-alpha.8/istioctl-1.6.0-
alpha.8-linux.tar.gz ... 
Failed. 

Trying with ISTIO_ARCH. Downloading istioctl-1.6.0-alpha.8 from https://github.com/istio/istio/releases/download/1.6.0-alpha.8/istioctl-1.6.0-alpha.8-linux-amd64.tar.gz ...
Download failed, please make sure your ISTIO_VERSION is correct and verify the download URL exists!
```

```
ISTIO_VERSION=1.6.0-alpha.2 ./downloadIstioCtl.sh
Downloading istioctl-1.6.0-alpha.2 from https://github.com/istio/istio/releases/download/1.6.0-alpha.2/istioctl-1.6.0-alpha.2-linux.tar.gz ... 
Failed. 

Trying with ISTIO_ARCH. Downloading istioctl-1.6.0-alpha.2 from https://github.com/istio/istio/releases/download/1.6.0-alpha.2/istioctl-1.6.0-alpha.2-linux-amd64.tar.gz ...
istioctl-1.6.0-alpha.2-linux-amd64.tar.gz download complete!

Add the istioctl to your path with:
  export PATH=$PATH:$HOME/.istioctl/bin 

Begin the Istio pre-installation verification check by running:
         istioctl verify-install 

Need more information? Visit https://istio.io/docs/reference/commands/istioctl/ 
```

```
./downloadIstioCtl.sh        
Downloading istioctl-1.5.1 from https://github.com/istio/istio/releases/download/1.5.1/istioctl-1.5.1-linux.tar.gz ... 
istioctl-1.5.1-linux.tar.gz download complete!

Add the istioctl to your path with:
  export PATH=$PATH:$HOME/.istioctl/bin 

Begin the Istio pre-installation verification check by running:
         istioctl verify-install 

Need more information? Visit https://istio.io/docs/reference/commands/istioctl/ 
```

File: release/downloadIstioCtl.sh [hunk 1]
```diff
@@ -43,22 +44,44 @@ if [ "x${ISTIO_VERSION}" = "x" ] ; then
...
+  printf "Failed. \n\nTrying with ISTIO_ARCH. Downloading %s from %s ...\n" "${NAME}" "$ARCH_URL"
+  if ! curl -fsLO "$ARCH_URL"
+  then
+   download_failed
+  else
+    filename="istioctl-${ISTIO_VERSION}-${OSEXT}-${ISTIO_ARCH}.tar.gz"
+    tar -xzf "${filename}"
+  fi
+else
+  filename="istioctl-${ISTIO_VERSION}-${OSEXT}.tar.gz"
```
Comment (carolynhu) [inline release/downloadIstioCtl.sh:79]: why the filename starts with `istioctl`, should be `istio`?

Comment (brian-avery) [reply release/downloadIstioCtl.sh:79]: The istio downloads start with `istio` and the istioctl downloads start with `istioctl`

File: release/downloadIstioCandidate.sh [hunk 2]
```diff
@@ -42,35 +42,57 @@ if [ "x${ISTIO_VERSION}" = "x" ] ; then
...
-  printf "Please verify the version you are trying to download.\n\n"
+  printf "Failed.\n\n Trying with ISTIO_ARCH. Downloading %s from %s ...\n" "$NAME" "$ARCH_URL"
+  if ! curl -fsLO "$ARCH_URL"
+  then
+    printf "\n\n"
+    printf "Unable to download Istio %s at this moment!\n" "$ISTIO_VERSION"
+    printf "Please verify the version you are trying to download.\n\n"
+    exit
+  else
+    filename="istio-${ISTIO_VERSION}-${OSEXT}-${ISTIO_ARCH}.tar.gz"
```
Comment (ericvn) [inline release/downloadIstioCandidate.sh:72]: This won't work for the files without the arch like: istio-1.6.0-alpha.2-osx.tar.gz

File: release/downloadIstioCandidate.sh [hunk 3]
```diff
@@ -42,35 +42,57 @@ if [ "x${ISTIO_VERSION}" = "x" ] ; then
 		  grep -v -E "(alpha|beta|rc)\.[0-9]$" | sort -t"." -k 1,1 -k 2,2 -k 3,3 -k 4,4 | tail -n 1)
 fi
 
+if [ "x${ISTIO_ARCH}" = "x" ]; then
+  ISTIO_ARCH=$(arch)
+fi
+
+#for now this supports amd64, which has arch output of x86_64. Thhe arch command will let us switch based on host
+ISTIO_ARCH="amd64"
+
```
Comment (ericvn) [inline release/downloadIstioCandidate.sh:51]: Why have the ability to set it if you are going to default anyway. Remove lines 45-48 as you aren't setting arch anyway, and it wouldn't matter if you did.

File: release/downloadIstioCandidate.sh (hunk 2 above)
Comment (ericvn) [reply release/downloadIstioCandidate.sh:72]: Set filename without the ISTIO_ARCH before the first curl, then update with ISTIO_ARCH inside the if?

File: release/downloadIstioCandidate.sh (hunk 3 above)
Comment (brian-avery) [reply release/downloadIstioCandidate.sh:71]: Updated to support multiple architectures. Copied from https://github.com/istio/istio/blob/abe49dfacd8b0c222e2dc2424fdad8187e01baf9/common/scripts/setup_env.sh#L28 with some changes to:
* validate user architecture entries
* allow users to enter either the `uname -m` name or common name for an architecture
* make shell check happy

File: release/downloadIstioCandidate.sh (hunk 2 above)
Comment (brian-avery) [reply release/downloadIstioCandidate.sh:92]: If I understand correctly, there's nothing to address here. Right?

Comment (ericvn) [reply release/downloadIstioCandidate.sh:92]: Correct

File: release/downloadIstioCtl.sh [hunk 4]
```diff
@@ -29,6 +29,7 @@ if [ "x${OS}" = "xDarwin" ] ; then
   OSEXT="osx"
 else
   OSEXT="linux"
+
```
Comment (ericvn) [inline release/downloadIstioCtl.sh:32]: Remove the blank line? Surprised linter didn't pick on this.

File: release/downloadIstioCtl.sh [hunk 5]
```diff
@@ -43,22 +44,64 @@ if [ "x${ISTIO_VERSION}" = "x" ] ; then
...
 cd "$tmp" || exit
 URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istioctl-${ISTIO_VERSION}-${OSEXT}.tar.gz"
-printf "Downloading %s from %s ... \n" "${filename}" "${URL}"
-curl -sLO "${URL}" || download_failed
+ARCH_URL="https://github.com/istio/istio/releases/download/${ISTIO_VERSION}/istioctl-${ISTIO_VERSION}-${OSEXT}-${ISTIO_ARCH}.tar.gz"
+
+printf "Downloading %s from %s ... \n" "${NAME}" "${URL}"
+if ! curl -fsLO "$URL"
+then
+  printf "Failed. \n\nTrying with ISTIO_ARCH. Downloading %s from %s ...\n" "${NAME}" "$ARCH_URL"
```
Comment (ericvn) [inline release/downloadIstioCtl.sh:90]: I don't think ISTIO_ARCH is correct as someone really can't set that value prior to calling the script, as it just gets set. You have TARGET_ARCH in the printf in the other file which is better. At least if we document that it's the `unmet -m` output that we are looking for (as opposed to amd64).

Comment (brian-avery) [reply release/downloadIstioCtl.sh:90]: Good catch. I fixed in the other file. Missed this one.

Comment (brian-avery) [reply release/downloadIstioCtl.sh:90]: From the user, the script will accept either x86_64, armv8, aarch64, armv, amd64, or arm64. Uname -m or common name for the arch :)

**Comment by ericvn** [comment]
Brian, thanks for this. I pulled both scripts and ran them with older and alpha.2 specified on my Mac and they worked OK. I had to remarked out the OS= line and passed in some OS=linux TARGET_ARCH=various with various releases and all that worked fine.

Since this script goes live as soon as it's merged,I would like at least a second person to review and approve before this goes in.

**Comment by shwsingh** [comment]
Since it was found during Internal dev testing @brian-avery pls add testing label

**Pull request merged**